bpf/bpf_lxc.c

```diff
@@ -956,11 +957,16 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
 		return DROP_INVALID;
 
 	/* Allow a hairpin packet to be redirected even if ENABLE_ROUTING is
-	 * disabled. Otherwise, the packet will be dropped by the kernel if
-	 * it is going to be routed via an interface it came from after it has
-	 * been passed to the stack.
+	 * disabled (e.g. the case in per-endpoint routes). Otherwise, the
+	 * packet will be dropped by the kernel if it is going to be routed via
+	 * an interface it came from after it has been passed to the stack.
+	 *
+	 * For the case where ENABLE_ROUTING is disabled, but the fast redirect
+	 * enabled we do lookup the local endpoint here to check whether we
+	 * must pass the packet up the stack for the host itself.
```

Comment:
That's not the only reason to lookup the local endpoint, no? We also want to run through `ipv4_local_delivery` to enforce ingress policies for that endpoint.